scarlet/interpolation.py

```diff
@@ -319,8 +365,44 @@ def sinc_interp(coord_hr, coord_lr, sample_lr):
     assert hy != 0
     assert hx != 0
 
-    return np.array([np.dot(np.dot(np.sinc((y_lr[np.newaxis, :]-y_hr[:, np.newaxis]) / hy),sample.T),
-                                        np.sinc((x_lr[:, np.newaxis]-x_hr[np.newaxis,:])/ hx) ) for sample in sample_lr])
+    if angle is None:
+        result = [np.dot(np.dot(np.sinc((y_lr[np.newaxis, :]-y_hr[:, np.newaxis]) / hy),image.T),
+                                        np.sinc((x_lr[:, np.newaxis]-x_hr[np.newaxis,:])/ hx) ) for image in images]
```

Comment:
Am I missing something here? You don't seem to be making use of the shifting trick from section 3.1 in the paper to limit the number of computations.

Reply:
Hehe indeed that made me scratch my head a little. Actually, here the splitting is implemented because the sinc is separable and we have one sinc that takes Nxshifts and another that takes Ny shifts and we never have to compute the NxxNy matrix, so it's done without having to go to Fourier. The other possible implementation is to write something like :
``` np.dot(np.sinc((y_lr[np.newaxis, :]-y_hr[:, np.newaxis]) / hy),shifted_image)```
Where `shifted_image` is the image that we transform in Fourier, apply the sinc and shift by y_hr Ny times, but this is actually about as long for small images and longer for large images because the ndim-fft doesn't handle these well. The problem here is that we go from low to high resolution unlike in render.